internal/util/merge/merge3.go

```diff
@@ -268,3 +272,74 @@ func metadataComment(node *yaml.RNode) string {
 	}
 	return mf.Key.YNode().LineComment
 }
+
+// resourceHandler is an implementation of the ResourceHandler interface from
+// kyaml. It is used to decide how a resource should be handled during the
+// 3-way merge. This differs from the default implementation in that if a
+// resource is deleted from upstream, it will only be deleted from local if
+// there is no diff between origin and local.
+type resourceHandler struct {
+	keptResources []*yaml.RNode
+}
+
+func (r *resourceHandler) Handle(origin, upstream, local *yaml.RNode) (filters.ResourceMergeStrategy, error) {
+	var strategy filters.ResourceMergeStrategy
+	switch {
+	case origin == nil && upstream == nil && local != nil:
+		strategy = filters.KeepDest
+	case upstream != nil && local == nil:
```

Comment:
What if upstream != nil && origin != nil && local == nil

This means the local resource is deleted. We should not write the resource from updated correct ?

Reply:
Good catch. Fixed this and added a test.